servicetalk-grpc-netty/src/test/java/io/servicetalk/grpc/netty/GrpcMessageEncodingTest.java

```diff
@@ -421,16 +413,19 @@ private TesterClient newClient(@Nullable final List<ContentCodec> supportedCodin
 
     @Test
     public void test() throws Throwable {
-        if (expectedSuccess) {
-            assertSuccessful(requestEncoding);
-        } else {
-            assertUnimplemented(requestEncoding);
+        try {
+            if (expectedSuccess) {
+                assertSuccessful(requestEncoding);
+            } else {
+                assertUnimplemented(requestEncoding);
+            }
+        } catch (Throwable t) {
+            throwAsyncErrors();
+            throw t;
         }
-
-        verifyNoErrors();
     }
 
-    private void verifyNoErrors() throws Throwable {
+    private void throwAsyncErrors() throws Throwable {
         if (!errors.isEmpty()) {
```

Comment:
nit: make this a concurrent queue to replace `empty` / `get(0)` (sensitive to race conditions) with `Object o = queue.poll()`

Reply:
Changed to keep consistent with other test classes. FWIW, there is no race here, the List was `synchronized` and the `throw` is called after the Futures are resolved.